### PR TITLE
Update tests to phpunit 8.5

### PR DESCRIPTION
**Pull request by stronk7**
Also
- Remove php70 tests because they don't support return :void
- Remove old versions using PHPUnit 6.x because they are missing the
  stringContainsString() assertion.
- Reorganize the travis files to avoid using exceptions and change
  to explicit positive alternative.

### CODE REVIEW

File: .travis.yml
Comment (sarjona) [inline .travis.yml:37]: Now that https://tracker.moodle.org/browse/MDL-69308 I guess it will be better to update the minimum PHP versions for master and 3.11.

**Review by sarjona — CHANGES_REQUESTED** [review]
Thanks for working on this Eloy. The patch looks good but, as 3.11 requirements issue landed yesterday, I think it will be better to update the PHP version to the minimum required for 3.11 and master.

BTW, I'm having issues to download locally your "pass" branch... I guess, for some reason that I don't know, git doesn't like this name (because, for instance, I can download and checkout perfectly others branches like "laptop"...).

**Comment by scara** [comment]
Hi @sarjona,
just lurking on your git issue, it works on my side:
```bash
$ git remote add stronk7 https://github.com/stronk7/moodle-local_moodlecheck
$ git remote -v
origin  https://github.com/scara/moodle-local_moodlecheck (fetch)
origin  https://github.com/scara/moodle-local_moodlecheck (push)
stronk7 https://github.com/stronk7/moodle-local_moodlecheck (fetch)
stronk7 https://github.com/stronk7/moodle-local_moodlecheck (push)
upstream        https://github.com/moodlehq/moodle-local_moodlecheck (fetch)
upstream        https://github.com/moodlehq/moodle-local_moodlecheck (push)
$ git fetch stronk7
$ git checkout --track stronk7/pass
Switched to a new branch 'pass'
Branch 'pass' set up to track remote branch 'pass' from 'stronk7'.
$ git log --format=%B -n 1 77050f154bbb5b22de64e9e421cc74bfb3f05070
Update tests to phpunit 8.5

Also
- Remove php70 tests because they don't support return :void
- Remove old versions using PHPUnit 6.x because they are missing the
  stringContainsString() assertion.
- Reorganize the travis files to avoid using exceptions and change
  to explicit positive alternative.
```
HTH,
Matteo

**Comment by sarjona** [comment]
Hi @scara !

Thanks for checking it! Maybe it depends on the git version because I'm still getting the same error:

```
$ git checkout --track eloy/pass
fatal: 'eloy/pass' is not a commit and a branch 'pass' cannot be created from it
sara@moodlelito:~/moodles/stable_master-codechecker/moodle/local/codechecker (master)$ git remote -v
eloy    https://github.com/stronk7/moodle-local_codechecker.git (fetch)
eloy    https://github.com/stronk7/moodle-local_codechecker.git (push)
github  git@github.com:sarjona/moodle-local_codechecker.git (fetch)
github  git@github.com:sarjona/moodle-local_codechecker.git (push)
origin  git://github.com/sarjona/moodle-local_codechecker.git (fetch)
origin  git://github.com/sarjona/moodle-local_codechecker.git (push)
upstream        git@github.com:moodlehq/moodle-local_codechecker.git (fetch)
upstream        git@github.com:moodlehq/moodle-local_codechecker.git (push)
victor  https://github.com/vmdef/moodle-local_codechecker.git (fetch)
victor  https://github.com/vmdef/moodle-local_codechecker.git (push)
$ git checkout --track eloy/pass
fatal: 'eloy/pass' is not a commit and a branch 'pass' cannot be created from it
$ git fetch https://github.com/stronk7/moodle.git pass
fatal: couldn't find remote ref pass
$ git --version
git version 2.29.2
```

I've been able to download other branches from Eloy in the past... And, in fact, I can download his laptop branch (that's why I thought it was due to the "pass" name of the branch because it might be some kind of "reserved" word or so...). But you've been able to download it, so there should be a different explanation for that!

**Comment by scara** [comment]
Hi @sarjona,
mine is:
```bash
$ git --version
git version 2.30.0.windows.1
```
You could try:
```bash
git fetch eloy
```
to update your local refs or just `git fetch eloy pass`.
In your command you're trying to search branch `pass` into Moodle core code 😉 i.e. just use your remote ref `eloy` here and you'll never fall into that typo.

HTH,
Matteo

**Comment by sarjona** [comment]
Hi Matteo!

While reviewing the issue I've been playing with fetch&friends... As you'll see, "pass" branch is not downloaded.
I'm using Ubuntu (so maybe that's something related to a reserved word for Ubuntu, not for git)

```
$ git fetch eloy
From https://github.com/stronk7/moodle-local_codechecker
 * [new branch]      MOODLE_21_STABLE -> eloy/MOODLE_21_STABLE
 * [new branch]      laptop           -> eloy/laptop
 * [new branch]      master           -> eloy/master
$ git fetch eloy pass
fatal: couldn't find remote ref pass
```

:-(

**Comment by stronk7** [comment]
git thinks you want to fetch my password! 🤣

File: .travis.yml
Comment (stronk7) [reply .travis.yml]: Done, I've moved all 311 and master tests from php72 to php73!

**Comment by stronk7** [comment]
Note I've added a 2nd commit here. It modifies slightly one of the existing tests to have #65 covered. Have done it here because in the other issue was going to conflict with this (modifying the same lines).

**Comment by stronk7** [comment]
OT, there is something in this GH review system that I don't get. I already applied the requested changes and resolved the conversation, but I continue seeing "1 review requesting changes -> sarjona requested changes".

Apparently there isn't way for me (well, as admin I can dismiss the request, but I won't) to say that the review changes have been performed, so I'm saying it now! 😄 

Ciao :-)

**Review by sarjona — APPROVED** [review]
Thanks for bumping the PHP version for 3.11 and master! The patch looks good so I'm happy to approve and merge it! :-)

**Comment by sarjona** [comment]
@stronk7 Hahahahahaha! Thanks for finding an explanation for the "pass" branch name mystery! 

The patch looks great and, as you'll see, I've been able to approve it. Regarding your comment about the "Request changes" behaviour, once I've approved changes, it has disappeared (but I agree is weird having this "request changes" there even when you're asking for a new review...).

**Comment by stronk7** [comment]
Thanks!

**Pull request merged**
